pkg/statser/statser_internal.go

```diff
@@ -0,0 +1,100 @@
+package statser
+
+import (
+	"context"
+	"time"
+
+	"github.com/atlassian/gostatsd"
+)
+
+// InternalHandler is an interface to dispatch metrics to.  Exists
+// to break circular dependencies.
+type InternalHandler interface {
+	DispatchMetric(ctx context.Context, m *gostatsd.Metric) error
+}
+
+// InternalStatser is a Statser which sends metrics to a handler
+type InternalStatser struct {
+	ctx       context.Context
```

Comment:
The convention is to not store context in fields but pass it as an argument i.e. it is a "per-request thing". Or when invoking a long running function. But sometimes it is inevitable and it feels like this is one of those cases.